azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/GraphOidcUser.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import java.util.Collection;
+import java.util.Map;
+
+public class GraphOidcUser implements OidcUser {
+    private final OidcUser delegate;
+    private final String graphApiToken;
+
+    public GraphOidcUser(OidcUser delegate, String token) {
```

Comment:
Could you please explain more about why `token` is required for `UserInfo` ?

Reply:
This was requested in order to be able to get/utilize the api token generated by calling/utilizing the spring boot security principal.

Reply:
from my understanding, `userinfo` should not contain any `token` like information. If you would like to get `token`, you should obtain that from SecurityContextHolder.

Reply:
so I assume the token to get from the SecurityContextHolder is as such:
`Authentication authentication = SecurityContextHolder.getContext().getAuthentication(); DefaultOidcUser principal = (DefaultOidcUser) authentication.getPrincipal(); 
OidcIdToken idToken = principal.getIdToken(); 
String accessToken = idToken.getTokenValue(); `

but I notice that this token gotten does not equal the api token, and also when trying to use this token against the graph.microsoft.com, get an access token validation failure.
might I be missing something/or does additional code need to be added in to set the graph api token in the SecurityContextHolder?

Reply:
I think the point is that if we should put `token` here, instead of the `token` validation failure, so we add the `token` to user info. Unless there is only way we can do, I may need to try your code later.